### PR TITLE
fix: minor `tsup` codegen fixes

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

Was playing around in [the sandbox](https://github.com/readmeio/tmp-api-tsup-sandbox) and had to make a few minor tweaks to get it up and running properly.

## 🧬 QA & Testing

Does the sandbox work?

### CODE REVIEW

File: packages/api/tsconfig.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "esModuleInterop": true,
     "lib": ["DOM", "DOM.Iterable", "ES2023"],
-    "outDir": "dist/"
+    "module": "NodeNext",
```
Comment (kanadgupta) [inline packages/api/tsconfig.json:5]: apparently JSON module import assertions don't support `module: Node16`. annoying!

File: packages/api/src/codegen/languages/typescript/index.ts [hunk 2]
```diff
@@ -167,6 +168,10 @@ export default class TSGenerator extends CodeGenerator {
   async compile(storage: Storage, opts: InstallerOptions = {}): Promise<void> {
     const installDir = storage.getIdentifierStorageDir();
 
+    await execa('npm', ['install', 'tsup', 'typescript', '-D'], {
```
Comment (kanadgupta) [inline packages/api/src/codegen/languages/typescript/index.ts:171]: we need TS as a dev-dep so the `tsup` build runs properly:

CleanShot 2023-10-16 at 15 09 46@2x

File: packages/api/package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "swagger"
   ],
   "dependencies": {
+    "@readme/api-core": "file:../core",
```
Comment (kanadgupta) [inline packages/api/package.json:41]: we grab the package version from `@readme/api-core`, which is why it's now a regular dep (but will defer to you on the best way to bring this in)

File: packages/api/tsconfig.json [hunk 4]
```diff
@@ -1,9 +1,10 @@
 {
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "esModuleInterop": true,
```
Comment (kanadgupta) [inline packages/api/tsconfig.json:4]: this is unnecessary, already enabled in the base config

File: packages/api/tsconfig.json [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
     "lib": ["DOM", "DOM.Iterable", "ES2023"],
-    "outDir": "dist/"
+    "module": "NodeNext",
+    "outDir": "dist/",
+    "resolveJsonModule": true
```
Comment (kanadgupta) [inline packages/api/tsconfig.json:7]: needed for the JSON import

File: packages/api/bin/api.js [hunk 6]
```diff
@@ -1,2 +1,2 @@
-#!/usr/bin/env node
+#!/usr/bin/env NODE_OPTIONS=--no-warnings node
```
Comment (kanadgupta) [inline packages/api/bin/api.js:1]: needed to silence this warning now that we're importing JSON modules

```
(node:16233) ExperimentalWarning: Import assertions are not a stable feature of the JavaScript language. Avoid relying on their current behavior and syntax as those might change in a future version of Node.js.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:16233) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
```

File: packages/api/src/codegen/languages/typescript/index.ts (hunk 2 above)
Comment (kanadgupta) [reply packages/api/src/codegen/languages/typescript/index.ts:171]: we probably want to include this in that list of packages along with some CLI explainers as to why we need said packages, but this PR is just intended to get it working properly

**Review by erunion — APPROVED** [review]

**Pull request merged**